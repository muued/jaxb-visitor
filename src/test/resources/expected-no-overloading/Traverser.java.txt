//
// This file was generated by the JavaTM Architecture for XML Binding(JAXB) Reference Implementation, v2.2.4 
// See <a href="http://java.sun.com/xml/jaxb">http://java.sun.com/xml/jaxb</a> 
// Any modifications to this file will be lost upon recompilation of the source schema. 
// Generated on: 2014.07.03 at 10:47:59 PM CEST 
//


package org.nooverload.visitor;

import jakarta.annotation.Generated;
import org.example.imported.ImportedData;
import org.example.imported.ImportedType;
import org.example.simple.ChoiceElement;
import org.example.simple.ComplexObject;
import org.example.simple.HasJAXBElement;
import org.example.simple.Problem;
import org.example.simple.Recursive;
import org.example.simple.TSimpleRequest;
import org.example.simple.TSimpleResponse;

@Generated("Generated by jaxb-visitor")
public interface Traverser<E extends Throwable >{


    void traverseImportedData(ImportedData aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverseImportedType(ImportedType aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverseMessage(ImportedType.Message aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverseChoiceElement(ChoiceElement aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverseComplexObject(ComplexObject aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverseLocalElement(ComplexObject.LocalElement aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverseHasJAXBElement(HasJAXBElement aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverseProblem(Problem aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverseRecursive(Recursive aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverseTSimpleRequest(TSimpleRequest aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverseTSimpleResponse(TSimpleResponse aBean, Visitor<?, E> aVisitor)
        throws E
    ;

}
