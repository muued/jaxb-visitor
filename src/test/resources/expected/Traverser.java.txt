//
// This file was generated by the JavaTM Architecture for XML Binding(JAXB) Reference Implementation, v2.2.4 
// See <a href="http://java.sun.com/xml/jaxb">http://java.sun.com/xml/jaxb</a> 
// Any modifications to this file will be lost upon recompilation of the source schema. 
// Generated on: 2014.07.03 at 10:47:59 PM CEST 
//


package org.example.visitor;

import jakarta.annotation.Generated;
import org.example.imported.ImportedData;
import org.example.imported.ImportedType;
import org.example.simple.ChoiceElement;
import org.example.simple.ComplexObject;
import org.example.simple.HasJAXBElement;
import org.example.simple.Problem;
import org.example.simple.Recursive;
import org.example.simple.TSimpleRequest;
import org.example.simple.TSimpleResponse;

@Generated("Generated by jaxb-visitor")
public interface Traverser<E extends Throwable >{


    void traverse(ImportedData aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverse(ImportedType aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverse(ImportedType.Message aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverse(ChoiceElement aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverse(ComplexObject aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverse(ComplexObject.LocalElement aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverse(HasJAXBElement aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverse(Problem aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverse(Recursive aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverse(TSimpleRequest aBean, Visitor<?, E> aVisitor)
        throws E
    ;

    void traverse(TSimpleResponse aBean, Visitor<?, E> aVisitor)
        throws E
    ;

}
